### PR TITLE
Set storage for BuildCommand and BuildEnvironment as private

### PR DESCRIPTION
**Pull request by stsewd**
This is how we have them in .com.

### CODE REVIEW

File: readthedocs/storage/s3_storage.py [hunk 1]
```diff
@@ -50,6 +50,10 @@ def __init__(self, *args, **kwargs):
                 'Ensure S3_BUILD_COMMANDS_STORAGE_BUCKET is defined.',
             )
 
+        self.bucket_acl = 'private'
+        self.default_acl = 'private'
+        self.querystring_auth = True
```
Comment (ericholscher) [inline readthedocs/storage/s3_storage.py:55]: We should make sure this works in dev, too.

**Review by ericholscher — APPROVED** [review]

**Comment by humitos** [comment]
We could update other PRs in .com to share these classes instead override them to also add the same parameters.

I didn't copied these attrs because i thought everything would be públic in org, but makes sense to have these private. So 👍

File: readthedocs/storage/s3_storage.py (hunk 1 above)
Comment (humitos) [reply readthedocs/storage/s3_storage.py:55]: We could move the `S3PrivateBucketMixin` I added in https://github.com/readthedocs/readthedocs-corporate/pull/1103/files#diff-ec70ada13c82176ac8d16446079e8f0966583f9d20ec9c1a80b3b95ae0d19fafR8-R17 to this repository and share this code with .com

**Review by humitos — APPROVED** [review]
Looks good. We can share more code with .com, tho

File: readthedocs/storage/s3_storage.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/storage/s3_storage.py:55]: yes, this works in dev. I think we can share a lot of code in .com now yes

**Pull request merged**